Document Data ingestion in the observability platform (#2731)
giantswarm/docs: src/content/overview/observability/data-management/data-ingestion/index.md

### Using ServiceMonitors

[ServiceMonitors](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml) are the primary way to configure metric collection for applications that expose metrics through a Kubernetes Service.

Here's a basic ServiceMonitor example:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    # Required for discovery by the metrics agent
    observability.giantswarm.io/tenant: my_tenant
    app.kubernetes.io/instance: my-service
  name: my-service
  namespace: monitoring
spec:
  endpoints:
  - interval: 60s   # Collection frequency
    path: /metrics  # Metrics endpoint path
    port: web       # Named port exposing metrics
  selector:         # Service label selector
    matchLabels:
      app.kubernetes.io/instance: my-service
```

### Using PodMonitors

[PodMonitors](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#podmonitorspec) are useful for collecting metrics directly from pods when a Service isn't necessary or doesn't exist.

Use PodMonitors when:

> - Your application doesn't require a Service for its primary function

Reviewer: ```suggestion

- Your application doesn't require a Service for its primary function
```

### Method 1: Pod labels (Recommended)

The simplest way to enable log collection is by adding a tenant label to your pods:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  namespace: example-namespace
spec:
  template:
    metadata:
      labels:
        # Enable automatic log collection
        observability.giantswarm.io/tenant: my_team
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
```

### Method 2: PodLogs resources

[PodLogs](https://grafana.com/docs/alloy/latest/reference/components/loki/loki.source.podlogs/#podlogs-custom-resource) provide more flexibility for advanced log collection scenarios:

```yaml
apiVersion: monitoring.grafana.com/v1alpha2
kind: PodLogs
metadata:
  name: example-podlog
  namespace: example-namespace
spec:
  namespaceSelector:
    matchLabels:
      kubernetes.io/metadata.name: example-namespace
  relabelings:
  # Configure tenant for data routing
  - action: replace
    replacement: myteam
    targetLabel: giantswarm_observability_tenant
  selector:
    matchLabels:
      app: nginx
```

Use PodLogs when you need to:

> - Filter pods using complex label selectors

Reviewer 2: ```suggestion

- Filter pods using complex label selectors
```

- Apply custom relabeling rules
- Collect from multiple namespaces
- Transform log metadata before ingestion

### Default log collection

System logs are automatically collected from these namespaces:
- `kube-system`
- `giantswarm`

These are managed through predefined PodLogs resources and shouldn't be modified.

## Multi-tenancy considerations

Both metrics and logs use tenant-based routing to ensure data isolation:

- **Metrics**: Use `observability.giantswarm.io/tenant` label on ServiceMonitors/PodMonitors
- **Logs**: Use `observability.giantswarm.io/tenant` pod label or `giantswarm_observability_tenant` relabeling in PodLogs

> **Important**: Data sent to non-existent tenants will be dropped. Ensure your tenant exists in a Grafana Organization before configuring data collection.

Reviewer: ```suggestion
**Important**: Data sent to non-existent tenants will be dropped. Make sure your tenant exists in a Grafana Organization before configuring data collection.
```
